backend/src/services/UserService.js

```diff
@@ -0,0 +1,28 @@
+import bcrypt from 'bcrypt';
+import { UserRepository } from '../repository/UserRepository';
+import User from '../models/User';
+
+export class UserService {
+  constructor() {
+    this.userRepository = new UserRepository();
+    console.log('UserService constructor');
+  }
+  async registerUser(user) {
+    const hashedPassword = await bcrypt.hash(user.password, 10);
+
+    let temp = [];
+    await this.userRepository.getUserData(user.name, user.email).then(data => {
+      temp = data.results;
+    });
```

Comment:
@marcianszlovak either use await / async or `then` callbacks but don't mix it
```suggestion
    let data = await this.userRepository.getUserData(user.name, user.email);
    temp = data.results;
```